webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/ncf/TechniqueReader.scala

```diff
@@ -53,11 +53,27 @@ class TechniqueReader(
     }
   }
 
+  def doesMethodsMetadataFileNeedsUpdate  ={
+    def getCfFilesInDir(file : File): Iterator[File] = {
+      if (file.isDirectory) {
+        file.children.flatMap(getCfFilesInDir)
+      } else {
+        if (file.extension(false).map(_ == "cf").getOrElse(false)) Iterator(file) else Iterator.empty
+      }
+    }
+    val baseDir = root / "usr" / "share" / "ncf" / "tree" / "30_generic_methods"
+    val userDir = configuration_repository / "ncf" / "30_generic_methods"
+
+    val methodsFileModifiedTime = methodsFile.lastModifiedTime()
+
+    (getCfFilesInDir(baseDir) ++ getCfFilesInDir(userDir)).map{_.lastModifiedTime()}.exists(_.isAfter(methodsFileModifiedTime))
```

Comment:
That seems inefficient. You could either collect directly max time when you iterate on childrens (map is not lazy, so you will go through all children for it), or even pass the comparison date in your collect's filter so that you just have to check for `iterator.nonEmpty` (which should stop on first match)